### DISCUSSION TITLE
CefSharp.DOM with a new WPF application and target framework NET 8.0

### DISCUSSION THREAD

**Discussion by ghost** [cefsharp/CefSharp.Dom#51] [Q&A] [answered]
Greetings,
I'm new to CefSharp and CefSharp.Dom but eager to learn. For a study project I started a new WPF application in Visual Studio 2022 (up to date) and target platform 8.0. After that used the commands install-package cefsharp.wpf.netcore and install-package cefsharp.dom.
So far so good. However when executing the line 'var devtoolContect = await browser.CreateDevToolsContextAsync() in Debug mode the processing seems to stop here. 
Which brings me to the question: Can CefSharp and CefSharp.Dom be used with .NET 8.0 or should I go back to an earlier version of .NET?
Hope to hear, Aad Slingerland

**Comment by amaitland** [comment, marked as answer]
I've not specifically tried `.Net 8`, so your best bet is to test it out using one of the pre-existing examples and changing to `net8.0-windows`

https://github.com/cefsharp/CefSharp.Dom/blob/main/lib/CefSharp.Dom.Wpf.Example/CefSharp.Dom.Wpf.Example.csproj#L5

**Reply by ghost** [reply]
And indeed I did study a number of samples. Got it working now. Thanks...

**Reply by mwpowellhtx** [reply]
_"How"_ (did you get it to work)? I am bumping into [_null exceptions_](https://github.com/cefsharp/CefSharp/discussions/4031#discussioncomment-9962413) around the `BrowserCore`. Maybe you were finding this as well 🤷‍♂️ ?

So far, my starting point is the [_minimal WPF sample_](https://github.com/cefsharp/CefSharp.MinimalExample/). Have not yet explored any DOM WPF samples, that sounds like my next destination, however.

**Reply by mwpowellhtx** [reply]
Got 'er sorted. Turns out my different await steps were more or less on point. The key is sorting out the correct selector query strings. As long as it works in the Chrome console, it should _"just work"_ here. There might be some trickery matching the generic types, is an array expected, and such as this, but otherwise, I found that selector string integrity was the key.
